src/plugins/intel_gpu/src/plugin/ops/parameter.cpp

```diff
@@ -167,96 +66,48 @@ static void CreateParameterOp(ProgramBuilder& p, const std::shared_ptr<ov::op::v
-            p.add_primitive(*op, cldnn::input_layout(batched_name, networkInputLayout));
+            std::string batched_name = input_name + suffix;
+            p.inputLayouts.insert({ op->get_friendly_name() + suffix, input_layout });
+            p.add_primitive(*op, cldnn::input_layout(batched_name, input_layout));
 
-            auto reorder_layout = networkInputLayout;
+            auto reorder_layout = input_layout;
             reorder_layout.format = cldnn::format::bfyx;
 
-            auto preprocessPrimID = "reorder:" + inputName + ProgramBuilder::m_preProcessTag + suffix;
-            auto reorder = cldnn::reorder(preprocessPrimID,
+            auto reorder_name = "reorder:" + input_name + ProgramBuilder::m_preProcessTag + suffix;
+            auto reorder = cldnn::reorder(reorder_name,
                                           cldnn::input_info(batched_name),
                                           reorder_layout);
             reorder.input_mem_type = cldnn::reorder::memory_type::surface;
             p.add_primitive(*op, reorder);
-            surfaces_inputs.push_back(cldnn::input_info(preprocessPrimID));
+            surfaces_inputs.emplace_back(reorder_name);
         }
 
         if (batch > 1 && !is_convert_color_input)
-            p.add_primitive(*op, cldnn::concatenation(inputName, surfaces_inputs, 0));
+            p.add_primitive(*op, cldnn::concatenation(input_name, surfaces_inputs, 0));
         else
-            p.primitive_ids[inputName] = "reorder:" + inputName + ProgramBuilder::m_preProcessTag;
-    } else if (is_convert_color_input) {
-        networkInputLayout.format = cldnn::format::byxf;
-
-        networkInputLayout.set_partial_shape({ input_pshape[0], input_pshape[3], input_pshape[1], input_pshape[2] });
-
-        p.inputLayouts.insert({ inputInfo->name(), networkInputLayout });
-        p.add_primitive(*op, cldnn::input_layout(inputName, networkInputLayout));
+            p.primitive_ids[input_name] = "reorder:" + input_name + ProgramBuilder::m_preProcessTag;
     } else {
-        auto preprocessPrimID = "reorder:" + inputName + ProgramBuilder::m_preProcessTag;
-        cldnn::layout inputLayout(networkInputLayout);
-        auto network_input_data_type = DataTypeFromPrecision(ip);
-        inputLayout.data_type = network_input_data_type;
-        p.inputLayouts.insert({ inputInfo->name(), inputLayout });
+        auto reorder_name = "reorder:" + input_name + ProgramBuilder::m_preProcessTag;
+        p.inputLayouts.insert({ op->get_friendly_name(), input_layout });
 
-        p.add_primitive(*op, cldnn::input_layout(inputName, inputLayout));
-
-        switch (preProcess.getMeanVariant()) {
-        case NONE: {
-            // If mean value is not specified and the data type does not change, do not add post reorder
-            if (network_input_data_type != networkInputLayout.data_type || connected_to_quantize(op)) {
```

Comment:
This checking code is added to prevent redundant reorder creation for performance optimization.
 Will you update codes to remove redundant reorder after this PR ?

Reply:
The original condition not to add unnecessary reorder is needed for performance, so I think it should be applied to this PR.

Reply:
Still don't we need the reorder if the input/output dtype are different?

Reply:
It can't be different anymore. 2.0 preprocessing is supposed to insert explicit Convert operation to the graph. So precision of input tensor must always match precision of `Parameter`

Reply:
Okay thanks for clarifying!